dot/rpc/subscription/websocket.go

```diff
@@ -364,6 +365,49 @@ func (c *WSConn) initRuntimeVersionListener(reqID float64, _ interface{}) (Liste
 	return rvl, nil
 }
 
+func (c *WSConn) initGrandpaJustificationListener(reqID float64, _ interface{}) (Listener, error) {
+	if c.BlockAPI == nil {
+		c.safeSendError(reqID, nil, "error BlockAPI not set")
+		return nil, fmt.Errorf("error BlockAPI not set")
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	jl := &GrandpaJustificationListener{
+		ctx:         ctx,
+		cancel:      cancel,
+		wsconn:      c,
+		subID:       c.qtyListeners,
```

Comment:
You need to take lock for `qtyListeners`

Reply:
Done!